editor.planx.uk/src/@planx/components/PlanningConstraints/Modal.tsx

```diff
@@ -0,0 +1,247 @@
+import Divider from "@mui/material/Divider";
+import Grid from "@mui/material/Grid";
+import Typography from "@mui/material/Typography";
+import visuallyHidden from "@mui/utils/visuallyHidden";
+import { Constraint, Metadata } from "@opensystemslab/planx-core/types";
+import omit from "lodash/omit";
+import React, { useState } from "react";
+import InputLabel from "ui/public/InputLabel";
+import ChecklistItem from "ui/shared/ChecklistItem";
+import ErrorWrapper from "ui/shared/ErrorWrapper";
+import Input from "ui/shared/Input";
+
+import { formatEntityName } from "./List";
+import { InaccurateConstraints } from "./Public";
+
+interface OverrideEntitiesModalProps {
+  showModal: boolean;
+  setShowModal: (value: React.SetStateAction<boolean>) => void;
+  fn: Constraint["fn"];
+  entities: Constraint["data"] | null;
+  metadata?: Metadata;
+  inaccurateConstraints: InaccurateConstraints;
+  setInaccurateConstraints: (
+    value: React.SetStateAction<InaccurateConstraints>,
+  ) => void;
+}
+
+const ERROR_MESSAGES = {
+  checklist: "Select at least one option",
+  input: "Enter a value",
+};
+
+export const OverrideEntitiesModal = ({
+  showModal,
+  setShowModal,
+  fn,
+  entities,
+  metadata,
+  inaccurateConstraints,
+  setInaccurateConstraints,
+}: OverrideEntitiesModalProps) => {
+  const [checkedOptions, setCheckedOptions] = useState<string[] | undefined>(
+    inaccurateConstraints?.[fn]?.["entities"],
+  );
+  const [showChecklistError, setShowChecklistError] = useState<boolean>(false);
+  const [textInput, setTextInput] = useState<string | undefined>(
+    inaccurateConstraints?.[fn]?.["reason"],
+  );
+  const [showInputError, setShowInputError] = useState<boolean>(false);
+
+  const title = `Which ${
+    metadata?.plural?.toLowerCase() || "entities"
+  } are inaccurate?`;
+
+  const closeModal = (_event: any, reason?: string) => {
+    if (reason && reason == "backdropClick") {
+      return;
+    }
+
+    // Clear any non-submitted inputs on cancel & sync parent state
+    setCheckedOptions(undefined);
+    setTextInput(undefined);
+    const newInaccurateConstraints = omit(inaccurateConstraints, fn);
+    setInaccurateConstraints(newInaccurateConstraints);
+
+    // Close modal
+    setShowModal(false);
+  };
+
+  const changeCheckbox =
+    (id: string) =>
+    (_checked: React.MouseEvent<HTMLButtonElement, MouseEvent> | undefined) => {
+      let newCheckedIds;
+
+      if (checkedOptions?.includes(id)) {
+        newCheckedIds = checkedOptions.filter((e) => e !== id);
+      } else {
+        newCheckedIds = [...(checkedOptions || []), id];
+      }
+
+      if (newCheckedIds.length > 0) {
+        setShowChecklistError(false);
+      }
+      setCheckedOptions(newCheckedIds);
+    };
+
+  const changeInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.target.value.length > 1) {
+      setShowInputError(false);
+    }
+    setTextInput(e.target.value);
+  };
+
+  const validateAndSubmit = () => {
```

Comment:
One edge-case workflow - 

- I mark a constraint as not applying
- I re-open modal to uncheck
- I can't save without error
- I have to cancel to get back to no overrides

This ties into the comment above on cancel behaviour - I think it is valid to submit this form with nothing checked, if the form is dirty.

Reply:
Can recreate and think I've got a solution for this one: 
- if you open an empty form and click "Submit", I'll still trigger the validation error messages and prevent submission
- if you open a pre-populated form, clear your selected options and text input, and click "Submit" - I'll sync your updates to parent state (essentially "clear") and close the modal
  - is it obvious you have to manually clear _both_ your selected options and text input? Would you expect text input to auto-clear as soon as zero checklist options are selected?

Reply:
Working as expected!

One very minor issue is that errors persist from the modal being closed and opened - I guess they should be cleared on both submit and cancel?